enterprise/metrics/src/main/java/org/neo4j/metrics/source/ClusterMetrics.java

```diff
@@ -0,0 +1,92 @@
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.metrics.source;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+
+import org.neo4j.function.Predicate;
+import org.neo4j.function.Predicates;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.ha.cluster.member.ClusterMember;
+import org.neo4j.kernel.ha.cluster.member.ClusterMembers;
+import org.neo4j.kernel.lifecycle.LifecycleAdapter;
+import org.neo4j.metrics.MetricsSettings;
+
+import static com.codahale.metrics.MetricRegistry.name;
+import static org.neo4j.kernel.ha.cluster.HighAvailabilityModeSwitcher.MASTER;
+import static org.neo4j.kernel.ha.cluster.HighAvailabilityModeSwitcher.UNKNOWN;
+
+/**
+ * Metrics reporting whether this cluster member IS_MASTER and whether or not this cluster member IS_AVAILABLE.
+ */
+public class ClusterMetrics extends LifecycleAdapter
+{
+    static final String NAME_PREFIX = "neo4j.cluster";
+    static final String IS_MASTER = name( NAME_PREFIX, "is_master" );
+    static final String IS_AVAILABLE = name( NAME_PREFIX, "is_available" );
+
+    private final MetricRegistry registry;
+    private final Config config;
+    private final ClusterMembers clusterMembers;
+
+
+    public ClusterMetrics( MetricRegistry registry, Config config, ClusterMembers clusterMembers )
+    {
+        this.registry = registry;
+        this.config = config;
+        this.clusterMembers = clusterMembers;
+    }
+
+    @Override
+    public void start() throws Throwable
+    {
+        if ( config.get( MetricsSettings.clusterEnabled ) )
+        {
+            registry.register( IS_MASTER, new RoleGauge( Predicates.equalTo( MASTER ) ) );
+            registry.register( IS_AVAILABLE, new RoleGauge( Predicates.not( Predicates.equalTo( UNKNOWN ) ) ) );
+        }
+    }
+
+    @Override
+    public void stop() throws Throwable
+    {
+        if ( config.get( MetricsSettings.clusterEnabled ) )
+        {
+            registry.remove( IS_MASTER );
+            registry.remove( IS_AVAILABLE );
+        }
+    }
+
+    private class RoleGauge implements Gauge<Integer>
+    {
+        private Predicate<String> rolePredicate;
+
+        public RoleGauge( Predicate<String> rolePredicate )
+        {
+            this.rolePredicate = rolePredicate;
+        }
+
+        public Integer getValue()
```

Comment:
why we converting result to integer here instead of just returning boolean?
Is it interface agreement that it should be always number type?